tests/test_multiannotator.py

```diff
@@ -0,0 +1,164 @@
+    n = sum(sizes)
+    local_data = []
+    labels = []
+    test_data = []
+    test_labels = []
+
+    for idx in range(m):
+        local_data.append(
+            np.random.multivariate_normal(mean=means[idx], cov=covs[idx], size=sizes[idx])
+        )
+        test_data.append(
+            np.random.multivariate_normal(mean=means[idx], cov=covs[idx], size=sizes[idx])
+        )
+        labels.append(np.array([idx for i in range(sizes[idx])]))
+        test_labels.append(np.array([idx for i in range(sizes[idx])]))
+    X_train = np.vstack(local_data)
+    true_labels_train = np.hstack(labels)
+    X_test = np.vstack(test_data)
+    true_labels_test = np.hstack(test_labels)
+
+    # Compute p(true_label=k)
+    py = np.bincount(true_labels_train) / float(len(true_labels_train))
+
+    noise_matrix = generate_noise_matrix_from_trace(
+        m,
+        trace=avg_trace * m,
+        py=py,
+        valid_noise_matrix=True,
+        seed=seed,
+    )
+
+    # Generate our noisy labels using the noise_matrix for specified number of annotators.
+    s = pd.DataFrame(
+        np.vstack(
+            [generate_noisy_labels(true_labels_train, noise_matrix) for _ in range(num_annotators)]
+        ).transpose()
+    )
+
+    # Each annotator only labels approximately 20% of the dataset
+    # (unlabeled points represented with NaN)
+    s = s.apply(lambda x: x.mask(np.random.random(n) < 0.8)).astype("Int32")
+    s.dropna(axis=1, how="all", inplace=True)
+
+    # s = generate_noisy_labels(true_labels_train, noise_matrix) # one annotator
+    # ps = np.bincount(s) / float(len(s))
+
+    # Compute inverse noise matrix
+    # inv = count.compute_inv_noise_matrix(py, noise_matrix, ps=ps)
+
+    # Estimate pred_probs
+    latent = count.estimate_py_noise_matrices_and_cv_pred_proba(
+        X=X_train,
+        labels=true_labels_train,
+        cv_n_folds=3,
+    )
+
+    # label_errors_mask = s != true_labels_train
+
+    row_NA_check = pd.notna(s).any(axis=1)
+
+    return {
+        "X_train": X_train[row_NA_check],
+        "true_labels_train": true_labels_train[row_NA_check],
+        "X_test": X_test[row_NA_check],
+        "true_labels_test": true_labels_test[row_NA_check],
+        "labels": s[row_NA_check].reset_index(drop=True),
+        # "label_errors_mask": label_errors_mask,
+        # "ps": ps,
+        # "py": py,
+        "noise_matrix": noise_matrix,
+        # "inverse_noise_matrix": inv,
+        # "est_py": latent[0],
+        # "est_nm": latent[1],
+        # "est_inv": latent[2],
+        # "cj": latent[3],
+        "pred_probs": latent[4][row_NA_check],
+        # "m": m,
+        # "n": n,
+    }
+
+
+# Global to be used by all test methods. Only compute this once for speed.
+data = make_data()
+
+
+# TODO: fix this test to work with NaN values - also unsure if this test should sit in this file
+# logic of current test does not work - randomly deleting data will break it (put in another file?)
```

Comment:
I think this test can live here since it's currently the only context where `_get_worst_class()` is ever used.